packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2822,26 +2822,56 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         // The call to fetch the snapshot is very expensive and not always needed.
         // It should only be done by the summarizerNode, if required.
         const snapshotTreeFetcher = async () => {
+            // When fetching from storage we will always get the latest version and do not use the ackHandle.
             const fetchResult = await this.fetchSnapshotFromStorage(
-                ackHandle,
-                summaryLogger,
+             null,
+             summaryLogger,
+             {
+                 eventName: "RefreshLatestSummaryGetSnapshot",
+                 ackHandle,
+                 summaryRefSeq,
+                 fetchLatest: true,
+             });
+             return fetchResult.snapshotTree;
+         };
+         const result = await this.summarizerNode.refreshLatestSummary(
+             proposalHandle,
+             summaryRefSeq,
+             snapshotTreeFetcher,
+             readAndParseBlob,
+             summaryLogger,
+         );
+
+         // In case we had to retrieve the latest snapshot and it is different than summaryRefSeq,
+         // wait for the delta manager to catch up.
+         if (result.latestSummaryUpdated === true && result.wasSummaryTracked === false) {
+            assert(result.snapshot !== undefined, "should have tree result");
+            const latestSnapshotRefSeq = await seqFromTree(result.snapshot, readAndParseBlob);
```

Comment:
The catch up has to happen before the summarizer node updates its state from the latest snapshot. Otherwise, the referenceSequenceNumber of the summarizer nodes will always be ahead of sequenceNumber of the catch up ops, and that is incorrect. For instance, `SummarizerNode::hasChanged()` will always return false until this catch up completes although at that point, the summarizer node did change.

Reply:
I must be missing something - I'm assuming the summarizer nodes will have the latest state and no matter what happens, once the catch up happens, we would be in a good state. The behavior is exactly the same as the one from when we call refreshLatestSummaryAckFromServer  in which we refresh the summarizer nodes and eventually wait for the deltamanager to catch up as well. Isn't it ?